cosmos/converter.py

```diff
@@ -372,3 +378,87 @@ def _add_dbt_project_hash_to_dag_docs(self, dag: DAG | None) -> None:
             logger.debug(f"Appended dbt project hash {dbt_project_hash} to DAG {dag.dag_id} documentation")
         except Exception as e:
             logger.warning(f"Failed to append dbt project hash to DAG documentation: {e}")
+
+    def _store_cosmos_telemetry_metadata_on_dag(  # noqa: C901
+        self,
+        dag: DAG | None,
+        render_config: RenderConfig,
+        project_config: ProjectConfig,
+        operator_args: dict[str, Any],
+        initial_load_method: LoadMode,
+    ) -> None:
+        """
+        Store Cosmos configuration metadata on the DAG for telemetry purposes.
+
+        This metadata is used by the DAG run listener to emit telemetry metrics
+        about how Cosmos is configured and used.
+
+        :param dag: The Airflow DAG to store metadata on. If None, no action is taken.
+        :param render_config: The render configuration
+        :param project_config: The project configuration
+        :param operator_args: The operator arguments
+        :param initial_load_method: The load method specified by the user (before automatic resolution)
+        """
+        if dag is None:
+            return
+
+        metadata = {}
+
+        # Compute each metric individually with error handling
+        try:
+            metadata["used_automatic_load_mode"] = initial_load_method == LoadMode.AUTOMATIC
+        except (AttributeError, TypeError) as e:
+            logger.warning(f"Failed to compute used_automatic_load_mode: {e}")
+
+        try:
+            metadata["actual_load_mode"] = str(self.dbt_graph.load_method.value)
+        except (AttributeError, TypeError) as e:
+            logger.warning(f"Failed to compute actual_load_mode: {e}")
+
+        try:
+            invocation_mode = None
+            if render_config.invocation_mode:
```

Comment:
If `render_config` is not `None`, then it will have an `invocation_mode` attribute:
https://github.com/astronomer/astronomer-cosmos/blob/0fa0163c02d709e0849e007e40f165ef0743d29c/cosmos/config.py#L86

The same applies to `dbt_deps` below:
https://github.com/astronomer/astronomer-cosmos/blob/0fa0163c02d709e0849e007e40f165ef0743d29c/cosmos/config.py#L90

And also:
- `uses_node_converter`
- `test_behavior`
- `source_behavior`

Would it be worth simplifying the code by checking whether `render_config` is `None` and setting those values accordingly, without the need of so many `try`/`except` statements?